ipc-queue/src/lib.rs

```diff
@@ -146,11 +146,13 @@ pub trait AsyncSynchronizer: Clone {
     fn notify(&self, event: QueueEvent);
 }
 
+#[allow(dead_code)]
 pub struct AsyncSender<T: 'static, S> {
     inner: Fifo<T>,
     synchronizer: S,
 }
 
+#[allow(dead_code)]
```

Comment:
Simply allowing dead code here, doesn't tell the user why this would be dead code. In this case it's only dead code when the target is not sgx. (The tests pass for the SGX target). I would annotate it as such. The same applies for the other errors as well I think.